From 0015686a9f6d020f8adb3d31a6f224ff24d28c82 Mon Sep 17 00:00:00 2001
From: Fei Yang <fei.yang@intel.com>
Date: Wed, 14 Nov 2018 18:00:38 -0800
Subject: [PATCH 0840/1214] Revert "REVERTME [IOTG]: drm/i915/: Add GuC v9.29
 and HuC v1.07 firmware size"

This reverts commit b13d30d990d678588614baa760851258f6ce1e73.

Tracked-On: OAM-71547
Signed-off-by: Badiuzzaman Iskhandar <badiuzzaman.azzarfan.bin.iskhandar@intel.com>
---
 drivers/gpu/drm/i915/intel_wopcm.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_wopcm.c b/drivers/gpu/drm/i915/intel_wopcm.c
index 10d1ea1..5ff877e 100644
--- a/drivers/gpu/drm/i915/intel_wopcm.c
+++ b/drivers/gpu/drm/i915/intel_wopcm.c
@@ -63,9 +63,6 @@
 #define GEN9_GUC_FW_RESERVED	(128 * 1024)
 #define GEN9_GUC_WOPCM_OFFSET	(GUC_WOPCM_RESERVED + GEN9_GUC_FW_RESERVED)
 
-#define GEN9_GUC_9_29_SIZE      ((142 * 1024) + 768)
-#define GEN9_HUC_1_07_SIZE      ((150 * 1024) + 576)
-
 /**
  * intel_wopcm_init_early() - Early initialization of the WOPCM.
  * @wopcm: pointer to intel_wopcm.
@@ -158,8 +155,8 @@ static inline int check_hw_restriction(struct drm_i915_private *i915,
 int intel_wopcm_init(struct intel_wopcm *wopcm)
 {
 	struct drm_i915_private *i915 = wopcm_to_i915(wopcm);
-	u32 guc_fw_size = GEN9_GUC_9_29_SIZE;
-	u32 huc_fw_size = GEN9_HUC_1_07_SIZE;
+	u32 guc_fw_size = intel_uc_fw_get_upload_size(&i915->guc.fw);
+	u32 huc_fw_size = intel_uc_fw_get_upload_size(&i915->huc.fw);
 	u32 ctx_rsvd = context_reserved_size(i915);
 	u32 guc_wopcm_base;
 	u32 guc_wopcm_size;
-- 
2.7.4

